R/module_teal_with_splash.R

```diff
@@ -65,17 +66,37 @@ srv_teal_with_splash <- function(id, data, modules, filter = teal_slices()) {
       shinyjs::showLog()
     }
 
-    is_pulled_data <- teal.data::is_pulled(data)
     # raw_data contains TealDataAbstract, i.e. R6 object and container for data
     # reactive to get data through delayed loading
     # we must leave it inside the server because of callModule which needs to pick up the right session
-    if (is_pulled_data) {
-      raw_data <- reactiveVal(data) # will trigger by setting it
+    raw_data <- if (inherits(data, "teal_data")) {
+      reactiveVal(data)
+    } else if (teal.data::is_pulled(data)) {
```

Comment:
just for clarity for future work - can you please add class condition so that it's clear that it's for `TealDataAbstract`

Reply:
Don't worry, I know every line in `teal`. It will be removed as soon as ddl touches main 👍